pkg/cmd/issue/edit/edit.go

```diff
@@ -157,59 +173,98 @@ func editRun(opts *EditOptions) error {
 		lookupFields = append(lookupFields, "milestone")
 	}
 
-	issue, repo, err := shared.IssueFromArgWithFields(httpClient, opts.BaseRepo, opts.SelectorArg, lookupFields)
+	// Get all specified issues and make sure they are within the same repo.
+	issues, repo, err := shared.IssuesFromArgsWithFields(httpClient, opts.BaseRepo, opts.SelectorArgs, lookupFields)
 	if err != nil {
 		return err
 	}
 
-	editable.Title.Default = issue.Title
-	editable.Body.Default = issue.Body
-	editable.Assignees.Default = issue.Assignees.Logins()
-	editable.Labels.Default = issue.Labels.Names()
-	editable.Projects.Default = append(issue.ProjectCards.ProjectNames(), issue.ProjectItems.ProjectTitles()...)
-	projectItems := map[string]string{}
-	for _, n := range issue.ProjectItems.Nodes {
-		projectItems[n.Project.ID] = n.ID
-	}
-	editable.Projects.ProjectItems = projectItems
-	if issue.Milestone != nil {
-		editable.Milestone.Default = issue.Milestone.Title
-	}
-
-	if opts.Interactive {
-		err = opts.FieldsToEditSurvey(&editable)
-		if err != nil {
-			return err
-		}
-	}
-
+	// Fetch editable shared fields once for all issues.
 	apiClient := api.NewClientFromHTTP(httpClient)
-	opts.IO.StartProgressIndicator()
+	opts.IO.StartProgressIndicatorWithLabel("Fetching repository information")
 	err = opts.FetchOptions(apiClient, repo, &editable)
 	opts.IO.StopProgressIndicator()
 	if err != nil {
 		return err
 	}
 
-	if opts.Interactive {
-		editorCommand, err := opts.DetermineEditor()
-		if err != nil {
-			return err
-		}
-		err = opts.EditFieldsSurvey(&editable, editorCommand)
-		if err != nil {
-			return err
-		}
+	// Update all issues in parallel.
+	if opts.Workers == 0 {
+		opts.Workers = 10
 	}
+	g, ctx := errgroup.WithContext(context.Background())
+	g.SetLimit(opts.Workers)
+
+	issueURLs := make([]string, 0, len(issues))
+	issuesChan := make(chan string)
+	doneCh := make(chan any)
+	go func() {
```

Comment:
Perhaps I am missing something but could this be simplified to not be run in a Go routine? It would make it so we don't need the extra `doneCh`.

Reply:
I didn't have this originally and didn't figure I needed it, but there was a race condition on my laptop where all the issue URLs were sent to the channel, but without signaling this `doneCh` they weren't all received when ranging over the `issueURLs` channel. I've used channels a lot and honestly couldn't figure out why, but having a separate channel like this certainly solves the problem. Any thoughts?

Reply:
I didn't see the code you had originally but if the `issuesChan` was unbuffered then I can see how it the code would deadlock on `g.Wait()` and also `issuesChan <- issue.URL`. If you haven't already could you try making `issueChan` with the length of the `issues` slice?